File: 프로그래머스/lv3/49191. 순위/순위.py
def solution(n, results):
    prev = [[False for _ in range(n+1)] for _ in range(n+1)]
    answer = 0
    
    for result in results:
        w,l = result
        prev[w][l] = True
    
    
    for k in range(1,n+1):
        for i in range(1,n+1):
            for j in range(1,n+1):
                if(prev[i][k] and prev[k][j]):
                    prev[i][j] = True
   
    for i in range(1,n+1):
        cnt = n -1
        for j in range(1,n+1):
            if(prev[i][j] or prev[j][i]):
                cnt -=1
    
        if(cnt == 0):
            answer += 1
            
    
    return answer

File: 프로그래머스/lv3/49191. 순위/순위.swift
import Foundation

func solution(_ n:Int, _ results:[[Int]]) -> Int {
    
    var prev:[[Bool]] = [[Bool]](repeating:Array(repeating:false,count:n+1),count:n+1)
    
    
    for result in results {
        
        let a = result[0]
        let b = result[1]
        
        prev[a][b] = true
    }
    
    var count:Int = 0
    
    for k in 1...n{
        for i in 1...n {
            
            for j in 1...n {
                
                if prev[i][k] && prev[k][j] { // i가 k이기고, k가 j이기면 
                    prev[i][j] = true // i가 j 이김
                }
                
            }
            
        }
    }
    
    
    for i in 1...n{
        
        var undecided:Int = n-1 // 자신은 제외
        
        for j in 1...n{
            
            
            
            if prev[i][j] || prev[j][i] { // 승패 하나라도 있으면 결정됨
                undecided -= 1
            }
            
        }
        
        if undecided == 0 { 
            count += 1 
        } //만약 정해지지 않은 상대가 없으면 , 정확한 순위를 매길 수 있음
    }
    
    
    return count
}